src/matcha_ml/templates/run_state_storage_template.py

```diff
@@ -0,0 +1,166 @@
+                "The file terraform.tfvars.json was not found in the "
+                f"current directory at {self.tfs.config.var_file}. Please "
+                "verify if it exists."
+            )
+            raise typer.Exit()
+
+    def _check_matcha_directory_exists(self) -> None:
+        """Checks if .matcha directory exists within the current working directory.
+
+        Raises:
+            typer.Exit: if the .matcha directory does not exist.
+            typer.Exit: if the .matcha directory does not contain the required files to deploy resources.
+        """
+        if not self.tfs.check_matcha_directory_exists():
+            print_error(
+                f"Error, the .matcha directory does not exist in {os.getcwd()} . Please ensure you are trying to destroy resources that you have provisioned in the current working directory."
+            )
+            raise typer.Exit()
+
+        if not self.tfs.check_matcha_directory_integrity():
+            print_error(
+                "Error, the .matcha directory does not contain files relating to deployed resources. Please ensure you are trying to destroy resources that you have provisioned in the current working directory."
+            )
+            raise typer.Exit()
+
+    def _initialize_terraform(self) -> None:
+        """Run terraform init to initialize Terraform .
+
+        Raises:
+            MatchaTerraformError: if 'terraform init' failed.
+        """
+        with Spinner("Initializing"):
+            ret_code, _, err = self.tfs.init()
+
+            if ret_code != 0:
+                print_error("The command 'terraform init' failed.")
+                raise MatchaTerraformError(tf_error=err)
+
+        print_status(
+            build_substep_success_status("Remote state management initialized!\n")
+        )
+
+    def _apply_terraform(self) -> None:
+        """Run terraform apply to create resources on cloud.
+
+        Raises:
+            MatchaTerraformError: if 'terraform apply' failed.
+        """
+        with Spinner("Applying"):
+            ret_code, _, err = self.tfs.apply()
+
+            if ret_code != 0:
+                raise MatchaTerraformError(tf_error=err)
+
+        print_status(
+            build_substep_success_status(
+                f"{Emojis.CHECKMARK.value} Your environment has been provisioned!\n"
+            )
+        )
+
+    def _get_terraform_output(self) -> Tuple[str, str, str]:
+        """Return the account name and the container name from terraform output.
+
+        Returns:
+            Tuple[str, str]: account name, the container name and azure resource_group_name.
+        """
+        tf_outputs = self.tfs.terraform_client.output()
+
+        account_name = ""
+        container_name = ""
+        resource_group_name = ""
+
+        for output_name, properties in tf_outputs.items():
+            property_name = output_name.replace("remote_state_storage_", "")
+            if property_name == "account_name":
```

Comment:
This logic can be simplified instead of using for loop.

```python
prefix = "remote_state_storage"
account_name = output_name[f"{prefix}_account_name"]["value"]
resource_group_name = output_name[f"{prefix}_resource_group_name"]["value"]
container_name = output_name[f"{prefix}_container_name"]["value"]
```